api/src/controllers/profile.ts

```diff
@@ -67,15 +73,20 @@ export const fetchAllProjectProfiles = async (req: Request, res: Response): Prom
 };
 
 export const fetchProjectProfile = async (
-  { params }: { params: any }, res: Response
+  { params, user }: { params: any, user: AuthenticatedUser }, res: Response
 ): Promise<void> => {
   const { ProfileModel } = dm;
   const { profileId } = params;
 
   try {
     const results = await ProfileModel.findById(Number(profileId));
 
-    res.status(200).json(results);
+    if (user.id === results.userId || user.roles.includes('administrator')) {
+      res.status(200).json(results);
+    } else {
+      throw errorWithCode('Unauthorized Access', 401);
```

Comment:
I don't think this error will make it out. It'll get caught by the catch below and replaced with `throw errorWithCode(message, 500)`; You probably need to check and re-throw.
```
if (err.code) {
  throw err
}
```